### PR TITLE
[SW-940] Add arm stow, carry, and ready services to spot_ros2

### PR DESCRIPTION
**Pull request by tcappellari-bdai**
## Change Overview

Added services for `stow_arm`, `ready_arm`, and `carry` to `spot_ros2`

## Testing Done

Ran the driver and tested the new services

Please create a checklist of tests you plan to do and check off the ones that have been completed successfully. Ensure that ROS 2 tests use `domain_coordinator` to prevent port conflicts. Further guidance for testing can be found on the [ros utilities wiki](https://github.com/bdaiinstitute/ros_utilities/wiki/Testing-guidelines).

### CODE REVIEW

**Comment by tcappellari-bdai** [comment]
* **#343** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/343?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tcappellari-bdai and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by bhung-bdai — APPROVED** [review]

File: spot_driver/spot_driver/spot_ros2.py [hunk 1]
```diff
@@ -572,6 +572,25 @@ def __init__(self, parameter_list: Optional[typing.List[Parameter]] = None, **kw
...
+        )
+        self.create_service(
+            Trigger,
+            "ready_arm",
+            lambda request, response: self.service_wrapper("ready_arm", self.handle_ready_arm, request, response),
+            callback_group=self.group,
+        )
+        self.create_service(
+            Trigger,
+            "carry",
```
Comment (khughes-bdai) [inline spot_driver/spot_driver/spot_ros2.py:589]: nit: `carry_arm`?

**Review by khughes-bdai — APPROVED** [review]
LGTM once CI passes. Is there a way to test what will happen if these services are called on a robot without an arm?

**Comment by tcappellari-bdai** [comment]
> LGTM once CI passes. Is there a way to test what will happen if these services are called on a robot without an arm?

oh shoot you're so right - totally forgot i meant to wrap these into an if statement

**Review by khughes-bdai — APPROVED** [review]

**Pull request merged**